src/os/bluestore/BlueStore.cc

```diff
@@ -4176,6 +4176,26 @@ void BlueStore::ExtentMap::init_shards(bool loaded, bool dirty)
   }
 }
 
+std::pair<uint32_t, uint32_t> BlueStore::ExtentMap::fault_range_ex(
+  KeyValueDB *db,
+  uint32_t offset,
+  uint32_t length)
+{
+  dout(30) << __func__ << " 0x" << std::hex << offset << "~" << length
+	   << std::dec << dendl;
+  if (shards.size() == 0) {
+    // no sharding yet; everyting is loaded
+    return {0, OBJECT_MAX_SIZE};
+  }
+  auto start = seek_shard(offset);
+  auto last = seek_shard(offset + length);
+  maybe_load_shard(db, start, last);
+  uint32_t left_bound = shards[start].shard_info->offset;
+  uint32_t right_bound =  (size_t)last < shards.size() ?
+                          shards[last].shard_info->offset : OBJECT_MAX_SIZE;
+  return {left_bound, right_bound};
```

Comment:
last shard is the shard with contain the offset+length, so last shard's offset should be < (offset + length), then right_bound will < (offset + length).  right_bound should = shards[last+1].shard_info->offset.